covid_graphs/covid_graphs/fit_atg_model.py

```diff
@@ -16,6 +16,9 @@ def predict(self, x: float) -> float:
         ys = _model(params=[self.a, self.tg, self.exp, self.t0], xs=np.array([x]))
         return ys[0]
 
+    def shift_forward(self, day_count: int) -> None:
```

Comment:
This feels hacky.

Reply:
Because the usage of `AtgModelFit` sucks! :smile: I think the implementation details (`t0`) leak and make the usage of `AtgModelFit` unnecessarily complicated.

To use the `predict` function, we have to do `func=lambda x: fit.predict(x + whole_day_offset)`! The funny thing is that the implementation immediately subtracts it away: `x + whole_day_offset - t0`. My change simplifies this non-sense, so that we only do `x - t0`.

However, I agree that that's hacky, but I don't have time to fix this file.

`t0` shouldn't be part of the final usage via `predict`. In the new proto I have start date and a variable called `offset`, which is less than 1 and together with start date marks the exact start of the curve. Basically start_date and offset together form `t0`.

Perhaps this file can get inspiration from it, but at best I can add a TODO now.